implementations/rust/ockam/ockam_core/src/routing/address.rs

```diff
@@ -139,6 +147,11 @@ impl Address {
     pub fn is_local(&self) -> bool {
         self.tt == LOCAL
     }
+
+    /// Take inner Vec
+    pub fn inner(self) -> Vec<u8> {
```

Comment:
Instead of exposing the implementation, I would add a `to_local` method and use it [here](https://github.com/build-trust/ockam/pull/8164/files#diff-fb6b2451268d57544e5b8570ee3d4cff294dab6325a3f48426547f219d83c452R145).

Reply:
I'm becoming a fan of exposing implementation lately to be honest. In context of rust it's a) allows to reduce number of clones in some places b) helps borrow checker figure out when we move the inner field, and not the whole struct. That being said I would even consider making fields public more often... the only thing I don't like, is that we also give mutable access to the field as a side-effect.
Speaking of this specific case, I don't think converting the Address to local makes any sense, it's only case - test code. So I vote against making that a part of public API

Reply:
I'm definitely not a fan of exposing implementations, especially not as a blanket rule. Performance optimisations can sometimes require it, but I'd rather have this trade-off on a case-by-case basis.